source/common/stats/raw_stat_data.cc

```diff
@@ -35,12 +37,48 @@ uint64_t RawStatData::structSizeWithOptions(const StatsOptions& stats_options) {
 
 void RawStatData::initialize(absl::string_view key, const StatsOptions& stats_options) {
   ASSERT(!initialized());
-  ASSERT(key.size() <= stats_options.maxNameLength());
+  ASSERT(key.length() <= stats_options.maxNameLength());
   ref_count_ = 1;
   memcpy(name_, key.data(), key.size());
   name_[key.size()] = '\0';
 }
 
+Stats::RawStatData* RawStatDataAllocator::alloc(absl::string_view name) {
+  // Try to find the existing slot in shared memory, otherwise allocate a new one.
+  Thread::LockGuard lock(mutex_);
+  if (name.length() > options_.maxNameLength()) {
+    ENVOY_LOG_MISC(
+        warn,
+        "Statistic '{}' is too long with {} characters, it will be truncated to {} characters",
+        name, name.size(), options_.maxNameLength());
+    name = name.substr(0, options_.maxNameLength());
+  }
+  auto value_created = stats_set_.insert(name);
+  Stats::RawStatData* data = value_created.first;
+  if (data == nullptr) {
+    return nullptr;
+  }
+  // For new entries (value-created.second==true), BlockMemoryHashSet calls Value::initialize()
+  // automatically, but on recycled entries (value-created.second==false) we need to bump the
+  // ref-count.
+  if (!value_created.second) {
+    ++data->ref_count_;
+  }
+  return data;
+}
+
+void RawStatDataAllocator::free(Stats::RawStatData& data) {
+  // We must hold the lock since the reference decrement can race with an initialize above.
+  Thread::LockGuard lock(mutex_);
+  ASSERT(data.ref_count_ > 0);
+  if (--data.ref_count_ > 0) {
```

Comment:
I know this is a copy/paste from the old version, but I intensely dislike the use of increment/decrement in the same statement as a comparison. Can you break this into two lines?

Reply:
I'm kind of OK with this as it's idiomatic in C but I'm fine with establishing an envoy-wide style guide outlawing this. Out of curiosity, WDYT of the pattern:
```
  a[i++] = 5;
  a[i++] = 7;
```
should that also be banned?

In any case I changed this one.

Reply:
Yes, I would be in favor of banning that pattern as well. But that's just my opinion, not the official style of the project. Feel free to leave this alone, as you just moved the lines without modification.